packages/flutter_tools/test/general.shard/android/gradle_test.dart

```diff
@@ -1704,6 +1704,56 @@ plugin1=${plugin1.path}
       ProcessManager: () => mockProcessManager,
     });
 
+    testUsingContext("gradle exit code is properly passed on", () async {
+      final File manifestFile = fileSystem.file('pubspec.yaml');
+      manifestFile.createSync(recursive: true);
+      manifestFile.writeAsStringSync('''
+        flutter:
+          module:
+            androidPackage: com.example.test
+        '''
+      );
+
+      fileSystem.file('.android/gradlew').createSync(recursive: true);
+
+      fileSystem.file('.android/gradle.properties')
+          .writeAsStringSync('irrelevant');
+
+      fileSystem.file('.android/build.gradle')
+          .createSync(recursive: true);
+
+      // Let any process start. Assert after.
+      when(mockProcessManager.run(
+        any,
+        environment: anyNamed('environment'),
+        workingDirectory: anyNamed('workingDirectory'),
+      )).thenAnswer((_) async => ProcessResult(1, 108, '', ''));
+
+      fileSystem.directory('build/outputs/repo').createSync(recursive: true);
+
+      await expectLater(() async =>
+        await buildGradleAar(
+          androidBuildInfo: const AndroidBuildInfo(BuildInfo(BuildMode.release, null, treeShakeIcons: false)),
+          project: FlutterProject.current(),
+          outputDirectory: fileSystem.directory('build/'),
+          target: '',
+          buildNumber: '1.0',
+        )
+      , throwsToolExit(exitCode: 108, message: 'Gradle task assembleAarRelease failed with exit code 108.'));
+
+      expect(
+        testLogger.statusText.contains('Consuming the Module'),
+        isFalse,
+      );
```

Comment:
maybe this expect is not necessary? if the exception is thrown, test succeeds?

Reply:
+1.  The test is about the exit code.

Reply:
fixed in https://github.com/flutter/flutter/pull/71582/commits/42d0c8dab16c728613e54bf394209cfbde7fdbc8

Reply:
Sorry for making you confused, I copied it from the above test, it actually does not need this expect.